packages/react-app/src/pages/[id]/edit.tsx

```diff
@@ -1,22 +1,47 @@
-import Error400 from '../400';
+import React from 'react';
+import { Heading, Stack } from '@chakra-ui/layout';
+import Layout from '@app/components/global/SiteLayout';
+import { useSession } from 'next-auth/react';
+import EditBountyForm from '@app/components/pages/Bounties/Form/EditBountyForm';
+import { useRouter } from 'next/router';
+import { useBounty } from '@app/hooks/useBounties';
+import { BountyLoader, BountyNotFound } from '.';
 
 const EditBounty = (): JSX.Element => {
-	// const router = useRouter();
-	// const { id, key } = router.query;
-	// const { data: bounty, error } = useSWR(
-	// 	id ? `/api/bounties/${id}?key=${key}` : null,
-	// 	fetcher
-	// );
-
-	// if (error) return <p>Failed to load</p>;
-	// if (!bounty) return <p>Loading...</p>;
-
+	const { status } = useSession({ required: false });
+	const unauthorized = (status !== 'loading' && status !== 'authenticated');
+	const router = useRouter();
+	const { id } = router.query;
+	const { bounty, isLoading, isError } = useBounty(id);
 	return (
-		<>
-			<Error400 />
-			{/* <Form bountyForm={bounty} /> */}
+		<>{
+			isLoading
+				// if loading, show the loader
+				? <BountyLoader />
+				: bounty
+					?
+					// if not loading and we have a bounty, show the page
+					<Layout title={unauthorized ? 'Not Authorized' : 'Edit Bounty'}>
+						{
+							unauthorized
+								?	(
+									<Stack align="center" justify="center" h="400px">
+										<Heading size="4xl" alignItems="center">
```

Comment:
Would it be possible to redirect to a predefined 403 page? Inspired thinking from some of your previous fantastic work in this CR